desc/optimize/optimizer.py

```diff
@@ -148,7 +147,7 @@ def optimize(  # noqa: C901 - FIXME: simplify this
         """
         if not isinstance(constraints, (tuple, list)):
             constraints = (constraints,)
-        things = flatten_list(things, flatten_tuple=True)
+        things = unique_list(flatten_list(things, flatten_tuple=True))[0]
```

Comment:
should maybe also check here that they didnt pass in duplicate things?

Reply:
I don't think duplicate things will actually break anything, but I added this check as a warning.